crates/services/bls_aggregation/src/bls_agg.rs

```diff
@@ -515,207 +522,255 @@ impl<A: AvsRegistryService + Send + Sync + Clone + 'static> BlsAggregatorService
-                };
-                // check if the operator has already signed for this digest
-                if aggregated_operators
-                    .get(&signed_digest.task_response_digest)
-                    .map(|operators| {
-                        operators
-                            .signers_operator_ids_set
-                            .contains_key(&signed_digest.operator_id)
-                    })
-                    .unwrap_or(false)
-                {
-                    signed_digest
-                        .result_channel
-                        .send(Err(BlsAggregationServiceError::SignatureVerificationError(SignatureVerificationError::DuplicateSignature)))
-                        .map_err(|_| BlsAggregationServiceError::ChannelError)?;
-                    continue;
-                }
+        let signed_digest =
+            signed_task_digest.ok_or(BlsAggregationServiceError::SignaturesChannelClosed)?;
+
+        // Verify if the operator has already signed for this digest
+        if Self::is_duplicate_signature(aggregated_operators, &signed_digest) {
+            signed_digest
+                .result_channel
+                .send(Err(BlsAggregationServiceError::SignatureVerificationError(
+                    SignatureVerificationError::DuplicateSignature,
+                )))
+                .map_err(|_| BlsAggregationServiceError::SenderError)?;
+            return Ok(());
+        }
 
-                    let verification_result = BlsAggregatorService::<A>::verify_signature(
-                        task_index,
-                        &signed_digest,
-                        &operator_state_avs,
-                        logger.clone(),
-                    )
-                    .await
-                    .map_err(BlsAggregationServiceError::SignatureVerificationError);
-                    let verification_failed = verification_result.is_err();
-
-                    signed_digest
-                        .result_channel
-                        .send(verification_result)
-                        .map_err(|_| BlsAggregationServiceError::ChannelError)?;
-
-                    if verification_failed {
-                        continue;
-                    }
+        // Verify the signature
+        let verification_result = verify_signature(
+            task_index,
+            &signed_digest,
+            operator_state_avs,
+            logger.clone(),
+        )
+        .await
+        .map_err(BlsAggregationServiceError::SignatureVerificationError);
+
+        // Send the verification result to the result channel
+        signed_digest
+            .result_channel
+            .send(verification_result.clone())
+            .map_err(|_| BlsAggregationServiceError::SenderError)?;
+
+        // If the signature is incorrect, return
+        let verification_has_error = verification_result.is_err();
+        if verification_has_error {
+            return Ok(());
+        }
```

Comment:
This is what I meant:

```suggestion
        let verification_has_error = verification_result.is_err();

        // Send the verification result to the result channel
        signed_digest
            .result_channel
            .send(verification_result)
            .map_err(|_| BlsAggregationServiceError::SenderError)?;

        // If the signature is incorrect, return
        if verification_has_error {
            return Ok(());
        }
```